File: Selenium/Project/Activity6.py
# Import webdriver from selenium
from selenium import webdriver
from selenium.webdriver.common.by import By

# Start the Driver
with webdriver.Firefox() as driver:
    # Navigate to the URL
    driver.get("https://alchemy.hguy.co/jobs")

    # Navigate to Jobs page
        
    driver.find_element(By.ID, "menu-item-24").click()
    page_heading = driver.find_element(By.ID, "menu-item-24")
    print("Page heading is: ", page_heading.text)
    
    # Print the title of the page
    print("Page title is: ", driver.title)
    
    driver.find_element(By.ID, "search_keywords").send_keys("Banking")
    driver.find_element(By.CLASS_NAME, "search_submit").click()
    driver.find_element(By.TAG_NAME, "H3").click()
    print("Page title is: ", driver.title)
    driver.find_element(By.XPATH, "//div/input[@class='application_button button']").click()
    get_email = driver.find_element(By.XPATH, "//div/p/a[@class='job_application_email']")
    print("Email ID is: ", get_email.text)

File: Selenium/Project/Activity6.java
package project;

import static org.testng.Assert.assertEquals;

import java.time.Duration;

import org.openqa.selenium.By;
import org.openqa.selenium.WebDriver;
import org.openqa.selenium.firefox.FirefoxDriver;
import org.openqa.selenium.support.ui.ExpectedConditions;
import org.testng.Assert;
import org.testng.annotations.AfterClass;
import org.testng.annotations.BeforeClass;
import org.testng.annotations.Test;
import org.openqa.selenium.support.ui.WebDriverWait;

public class Activity6 {
	//Declare webdriver
	WebDriver driver;
	WebDriverWait wait = new WebDriverWait(driver, Duration.ofSeconds(10));
	//Set up method
	@BeforeClass
	  public void setUp() {
		   {
		        // Initialize webdriver
		        driver = new FirefoxDriver();
		        //Open the page
		        driver.get("https://alchemy.hguy.co/jobs");
		  }
	  }
	
	
  @Test(priority = 1)
  public void jobsPageTest() {
	  driver.findElement(By.id("menu-item-24")).click();
	  String jobsheading = driver.findElement(By.id("menu-item-24")).getText();
	  System.out.println("Page Title " + jobsheading);
	  System.out.println("Page Title " + driver.getTitle());
      assertEquals(driver.getTitle(), "Jobs – Alchemy Jobs");
      
      driver.findElement(By.id("search_keywords")).sendKeys("Banking");
      driver.findElement(By.className("search_submit")).click();
      
     // wait.until(ExpectedConditions.elementToBeClickable(By.tagName("H3")));
   //   driver.findElement(By.className(contains(text,"IBM"))).click();
      driver.findElement(By.tagName("H3")).click();
      System.out.println("Page Title " + driver.getTitle());
  //    wait.until(ExpectedConditions.elementToBeClickable(By.xpath("//div[@class='job_application application']")));
      driver.findElement(By.xpath("//div/input[@class='application_button button']")).click();
      String getEmailId = driver.findElement(By.xpath("//div/p/a[@class='job_application_email']")).getText();
      System.out.println("Email ID " + getEmailId);
   
  }
  
    
  @AfterClass
  public void tearDown() {
      // Close the browser
      driver.quit();
  }
  
}
